src/eos-updater-apply.c

```diff
@@ -103,14 +111,17 @@ apply (GTask *task,
     goto error;
 
   bootversion = ostree_sysroot_get_bootversion (sysroot);
-  merge_deployment = ostree_sysroot_get_merge_deployment (sysroot, NULL);
+  booted_deployment = eos_updater_get_booted_deployment_from_loaded_sysroot (sysroot,
+                                                                             &error);
+  if (booted_deployment == NULL)
+    goto error;
```

Comment:
That said, I'm not sure of a clean way to implement the stuff that's in the `error` label using something like `g_autoptr`.